include/fmt/std.h

```diff
@@ -389,6 +391,52 @@ struct formatter<
 #endif
   }
 };
+
+namespace detail {
+
+template <typename T, typename Enable = void>
+struct has_flip : std::false_type {};
+
+template <typename T>
+struct has_flip<T, void_t<decltype(std::declval<T>().flip())>>
+    : std::true_type {};
+
+template <typename T> struct is_bit_reference_like {
+  static constexpr const bool value =
+      std::is_convertible<T, bool>::value &&
+      std::is_nothrow_assignable<T, bool>::value && has_flip<T>::value;
+};
+
+}  // namespace detail
+
+FMT_EXPORT
+template <typename BitRef, typename Char>
+struct formatter<BitRef, Char,
+                 enable_if_t<detail::is_bit_reference_like<BitRef>::value>>
```

Comment:
I don't think we need SFINAE here. Let's use `std::vector<bool>::reference`.

Reply:
1) The compiler won't be deduce the allocator type from ``std::vector<bool, Allocator>::reference`` or ``N`` value from ``std::bitset<N>::reference``.

2) What is ``std::vector<bool>::reference`` or ``std::bitset<N>::reference``?
On libc++ is ``std::__bit_reference`` and ``std::__bit_const_reference``.
On gcc and msvs ``std::vector<bool>::reference`` and ``std::bitset<N>::reference`` - is different proxy classes.

We need separate implementation for libc++ and gcc/msvc. But in this case, we will not be able to implement output for ``std::bitset``.

See:
https://github.com/fmtlib/fmt/pull/3570#discussion_r1284995184 (Allocator)
https://github.com/fmtlib/fmt/pull/3570#discussion_r1285057473 (const)

Previous branch:
https://github.com/fmtlib/fmt/tree/85bc49b83d59070a8023f529ec80c38652c14910

Commit with formatters: https://github.com/fmtlib/fmt/commit/64c905e27868c246289fd5d5e13663cc7410cfce
Failed tests: https://github.com/fmtlib/fmt/actions/runs/5769810223/job/15642249348 , https://github.com/fmtlib/fmt/actions/runs/5769810222/job/15642249612

Reply:
Why do we care about allocator? Reference type should normally be the same between vectors with different allocators: https://godbolt.org/z/5sKM5b6vb.

Reply:
No. It is implementation specific behaviour:
libc++: https://godbolt.org/z/3xbxj89hn

Reply:
That's unfortunate.

Reply:
A short comment explaining why we can't use `vector<bool>::reference` would be nice though.

Reply:
Done.